extensions/ext/GLSL_EXT_shader_tile_image.txt

```diff
@@ -0,0 +1,306 @@
+    uattachmentEXT   a handle for accessing an unsigned integer color framebuffer attachment
+
+    Modify the Definition subsection of 4.1.7, Opaque types from
+
+      They can only be declared as function parameters or in uniform-qualified
+      variables (see "Uniform Variables").
+
+    to
+
+      With the exception of attachment types (see "Attachment Types"), they can
+      only be declared as function parameters or in uniform-qualified variables
+      (see "Uniform Variables").
+
+    Add sub section under Section 4.1.7, Opaque types
+
+    4.1.7.x Attachment Types
+
+    Attachment types are only available when targeting Vulkan.
+
+    Attachment types (e.g., attachmentEXT) are opaque types, declared and
+    behaving as described above for opaque types.  When aggregated into arrays
+    within a shader, they can only be indexed with a dynamically uniform
+    integral expression, otherwise results are undefined.
+
+    Attachment types can only be declared as function parameters or tileImageEXT
+    qualified variables.
+
+    Attachment types are used to read framebuffer color attachment values from
+    within fragment shaders using attachment read functions listed in section
+    8.x, Attachment read functions.  Depth and stencil attachment values are
+    read without explicit attachment variables.
+
+    Attachment types are only available in fragment shaders. It is a
+    compile-time error to use them in any other stage. It is a compile-time error
+    to declare both attachment type and subpass-input type variables in a shader.
+
+    Members of structures cannot be declared with attachment types.
+
+    Modify Section 4.3, Storage Qualifiers (p. 46)
+
+    (add to storage qualifier table, p. 47)
+
+    Storage Qualifier    Meaning
+    -----------------    ------------------------------------------
+    tileImageEXT         fragment shader only; attachment variables with
+                         tileImageEXT storage can be used to access the
+                         framebuffer value at a pixel location.
+
+    Add a subsection to Section 4.3, Storage Qualifiers
+
+    4.3.x Tile Image Variables
+
+    Tile image variables are only available when targeting Vulkan.
+
+    The tileImageEXT qualifier is used to declare global variables of type
+    attachmentEXT, iattachmentEXT, or uattachmentEXT in fragment shaders. It is a
+    compile time error to use the qualifier with any other types or in any other
+    shader stage.
+
+    Modify Section 4.4, Layout Qualifiers (p. 63)
+
+    (add to the layout qualifier table, pg. 63-65)
+
+                                              Qualifier  Individual           Block     Allowed
+    Layout Qualifier                            Only      Variable   Block   Member   Interfaces
+    ----------------------------------------  ---------  ----------  -----  --------  -----------
+    non_coherent_color_attachment_readEXT        X          -         -        -     fragment in
```

Comment:
It has been suggested that the allowed interfaces here should instead be 'tileImageEXT'.

That would require some minor changes to the glslang implementation and (Khronos internal) CTS patches. I have not yet made this change, though could be persuaded to.